RustLeanModels/Basic.lean

```diff
@@ -100,11 +120,11 @@ lemma map_sub_add {l: List Nat} (g: ∀ x ∈ l, x ≥ c): List.map (fun x => x
   have: ∀ m ∈ t, m ≥ c := by simp only [mem_cons, ge_iff_le, forall_eq_or_imp] at g; exact g.right
   exact ind this
 
-lemma map_add_comm {l: List Nat} : List.map (fun x => x + a) (List.map (fun x => x + b) l) = List.map (fun x => x + b) (List.map (fun x => x + a) l) :=by
-  simp only [map_map, comp_add_right]; have: (a+b) = (b+a) := by omega
-  rw[this]
+theorem map_add_comm {l: List Nat} : List.map (fun x => x + a) (List.map (fun x => x + b) l) = List.map (fun x => x + b) (List.map (fun x => x + a) l) :=by
+  simp only [map_map, map_inj_left, Function.comp_apply]
+  omega
 
-lemma map_sub_comm : List.map (fun x: Nat=> x-a) (List.map (fun x=> x-b) l) = List.map (fun x=> x-b) (List.map (fun x=> x-a) l) :=by
+theorem map_sub_comm : List.map (fun x: Nat=> x-a) (List.map (fun x=> x-b) l) = List.map (fun x=> x-b) (List.map (fun x=> x-a) l) :=by
```

Comment:
The proof of `map_add_com` also works for `map_sub_comm` (and it's simpler).

Reply:
Updated.